File: trunk/xml-only/src/dataModel/Dataset.cpp
/* 
 * File:   Dataset.cpp
 * Author: jolo
 * 
 * Created on 10. Dezember 2009, 19:41
 */

#include "Dataset.h"

#include <QObject>
#include <QStringList>

Dataset::Dataset(int id,
        const QString& name,
        const QString& description)
  : PrimitiveDataset(id),
_name(name),
_description(description) {
}

Dataset::~Dataset() {
}

QString Dataset::makeToolTipText(const bool withDatasetName) const {
    QStringList ret;
    if (withDatasetName) {
        ret << QObject::tr("Dataset:");
    }

    ret << QObject::tr("ID: %1").arg(getId())
            << QObject::tr("Name: %1").arg(getName())
            << QObject::tr("Description: %1").arg(getDescription());
    return ret.join("\n");
}

void Dataset::copyData(Dataset* other) {
  setName(other->getName());
  setDescription(other->getDescription());
}

File: trunk/src/Model/Dataset.py
class Dataset(object):
    id = None
    def __init__(self, id=None):
        object.__init__(self)
        self.id = id
    def hasId(self):
        return self.id is not None
    def __str__(self):
        return u"<%1s: ID: %i>" % (self.__class__.__name__, self.id)
    def makeToolTip(self):
        return u'ID: %i' % self.id
